core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -501,7 +501,7 @@ public void testNewTableMetadataReassignmentAllIds() throws Exception {
 
     PartitionSpec spec = PartitionSpec.builderFor(schema).withSpecId(5)
         .add(3, 1005, "x_partition", "bucket[4]")
-        .add(5, 1005, "z_partition", "bucket[8]")
+        .add(5, 1003, "z_partition", "bucket[8]")
```

Comment:
This was a typo before?

Reply:
The intent is to check whether the IDs are reassigned. The spec builder may reject duplicate IDs.